python/triton/runtime/build.py

```diff
@@ -40,11 +40,12 @@ def _build(name, src, srcdir, library_dirs, include_dirs, libraries):
     if scheme == 'posix_local':
         scheme = 'posix_prefix'
     py_include_dir = sysconfig.get_paths(scheme=scheme)["include"]
-    include_dirs = include_dirs + [srcdir, py_include_dir]
+    custom_backend_dirs = set(os.getenv(var) for var in ('TRITON_CUDACRT_PATH', 'TRITON_CUDART_PATH', 'TRITON_CUPTI_PATH'))
```

Comment:
why do we include CUPTI here? 

The reason why triton builds with its own cuda is to avoid supporting multiple version. In what case is this causing a problem?

Reply:
I think the author of the PR is trying to use their own CUDA version instead of the version shipped with triton

Reply:
We build our docker images from nvidia/cuda images, which already have cuda toolkit installed. We tried to build Triton with the latest cuda toolkit included in the docker images and have consistent cuda runtime versions.

Sorry if the CUPTI path is redundant. In nvidia/cuda images, all three of `('TRITON_CUDACRT_PATH', 'TRITON_CUDART_PATH', 'TRITON_CUPTI_PATH')` variables actually point to the same `/usr/local/cuda/include` path. I guess it doesn't hurt to include all of those (with `set` to deduplicate), but feel free to remove any unnecessary items.

Reply:
done

Reply:
> I think the author of the PR is trying to use their own CUDA version instead of the version shipped with triton

The part that confuse me is that the paths are only use for include. I don't understand what part would break from using different includes than the runtime used. 
We use triton with different runtime than the headers we compiled with all the time. 
What kind of error are you getting?

Reply:
> What kind of error are you getting?

I already described that in https://github.com/triton-lang/triton/issues/4224. It's 
```
/tmp/tmpqiqtcxvu/main.c:1:10: fatal error: cuda.h: No such file or directory
    1 | #include "cuda.h"
      |          ^~~~~~~~
compilation terminated.
```

> We use triton with different runtime than the headers we compiled with all the time.

I don't quite follow this. It's simply because setting `TRITON_CUDART_PATH` skips downloading headers to `py_include_dir` at Triton compile time, but Triton doesn't include `TRITON_CUDART_PATH` at runtime.

Reply:
> I don't quite follow this. I think it's because setting TRITON_CUDART_PATH skips downloading headers to py_include_dir at Triton build time, but Triton doesn't include TRITON_CUDART_PATH at runtime.

ok missed that, makes sense